micrometer-core/src/main/kotlin/io/micrometer/core/instrument/kotlin/ObservationCoroutineContextServerInterceptor.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2013-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.micrometer.core.instrument.kotlin
+
+import io.grpc.Metadata
+import io.grpc.ServerCall
+import io.grpc.kotlin.CoroutineContextServerInterceptor
+import io.micrometer.observation.ObservationRegistry
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * This interceptor is meant to propagate observation context to a kotlin coroutine gRPC server method.
+ *
+ * Usage:
+ *
+ * ```
+ * val server = ServerBuilder.forPort(8080)
+ *         .intercept(ObservationCoroutineContextServerInterceptor(observationRegistry))
+ *         .intercept(ObservationGrpcServerInterceptor(observationRegistry))
+ *         .build();
+ * server.start()
+ * ```
+ *
+ * Please remember that order of interceptors matters, and it has to be the same as it is in the example above.
+ *
+ * @since ??? TODO should we set this property?
+ */
+class ObservationCoroutineContextServerInterceptor(
```

Comment:
I wonder how to document this change, and where to put example code? WDYT?

Reply:
What about docs? Shouldn't we mention how to set it up for kotlin gRPC services that use coroutines? 🙂

Reply:
I've set `since` property and made a new commit - I guess that we would squash them before merge? 🙂